library/src/main/java/com/yuyh/library/EasyGuide.java

```diff
@@ -1,384 +1,416 @@
+            params.setMarginRight(-offsetX);
         } else {
-            params.leftMargin = offsetX;
+            params.setMarginLeft(offsetX);
         }
-
         if (offsetY == Constants.CENTER) {
-            params.addRule(RelativeLayout.CENTER_VERTICAL, RelativeLayout.TRUE);
+            params.addRule(DependentLayout.LayoutConfig.VERTICAL_CENTER, DependentLayout.LayoutConfig.TRUE);
         } else if (offsetY < 0) {
-            params.addRule(RelativeLayout.ALIGN_PARENT_BOTTOM, RelativeLayout.TRUE);
-            params.bottomMargin = -offsetY;
+            params.addRule(DependentLayout.LayoutConfig.ALIGN_PARENT_BOTTOM, DependentLayout.LayoutConfig.TRUE);
+            params.setMarginBottom(-offsetY);
         } else {
-            params.topMargin = offsetY;
+            params.setMarginTop(offsetY);
         }
-
-        mGuideView.addView(view, params);
+        mGuideView.addComponent(view, params);
     }
 
+    /**
+     * Is dialog is showing or not.
+     */
     public boolean isShowing() {
-        return mParentView.indexOfChild(mGuideView) > 0;
+        return mParentView.getChildIndex(mGuideView) > 0;
     }
 
-    public boolean inRangeOfView(View view, MotionEvent ev) {
-        int[] location = new int[2];
-        view.getLocationOnScreen(location);
+    private boolean inRangeOfView(Component view, TouchEvent ev) {
+        int[] location = view.getLocationOnScreen();
         int x = location[0];
-        int y = location[1];
-        if (ev.getX() < x || ev.getX() > (x + view.getWidth()) || ev.getY() < y || ev.getY() > (y + view.getHeight())) {
-            return false;
-        }
-        return true;
+        int y = location[1] - 130;
+        MmiPoint point = ev.getPointerScreenPosition(0);
+        return point.getX() >= x && point.getX() <= (x + view.getWidth())
+                && point.getY() >= y && point.getY() <= (y + view.getHeight());
     }
 
     public static class Builder {
 
-        Activity activity;
+        AbilitySlice activity;
+
+        StackLayout componentContainer;
 
         List<HighlightArea> areas = new ArrayList<>();
+
         List<TipsView> views = new ArrayList<>();
+
         List<Message> messages = new ArrayList<>();
 
         Confirm confirm;
 
         boolean dismissAnyWhere = true;
+
         boolean performViewClick;
 
-        public Builder(Activity activity) {
+        public Builder(AbilitySlice activity) {
             this.activity = activity;
         }
 
         /**
-         * 添加高亮区域
+         * Add Highlight area.
          *
-         * @param view
-         * @param shape 高亮区域形状
-         * @return
+         * @param view view
+         * @param shape shape
          */
-        public Builder addHightArea(View view, @HShape int shape) {
+        public Builder addHightArea(Component view, int shape) {
             HighlightArea area = new HighlightArea(view, shape);
             areas.add(area);
             return this;
         }
 
```

Comment:
public Builder addHightLightArea(HighlightArea area); - public method api is missing in hmos code. Please check